File: web/src/models/Post.ts
import User from "./User";

export default class Post {
    id: string | null;
    user: User;
    text: string;
    date: number;
    reactions: Map<string, number> | null;
    
    constructor(id: string | null, user: User, text: string, date: number, reactions: Map<string, number> | null) {
        this.id = id;
        this.user = user;
        this.text = text;
        this.date = date;
        this.reactions = reactions;
    }

    static fromJson(object: any): Post {
        const map: Map<string, number> = new Map();
        Object.entries(object.reactions || {}).forEach((entries: Array<any>) => {
            const key: string = entries[0];
            map.set(key, entries[1]);
        });
        return new Post(object.id,
            object.user,
            object.text,
            object.date,
            map,
        )
    }
}

File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/post/Post.kt
package com.adqmobile.socialnetwork.domain.post

import androidx.room.*
import com.adqmobile.socialnetwork.data.local.DatabaseMapConverter
import com.adqmobile.socialnetwork.domain.user.User
import com.google.gson.annotations.SerializedName
import java.util.*

@Entity(tableName = "posts")
data class Post(@PrimaryKey @field:SerializedName("id") val id: Long?,
                @Embedded(prefix = "user_") val user: User,
                @field:SerializedName("text") val text: String,
                @field:SerializedName("date") val date: Long,
                @field:SerializedName("reactions") @TypeConverters(DatabaseMapConverter::class) val reactions: Map<String, Int>?) {

    fun formattedDate(): String {
        return Date(date).toGMTString()
    }
}